awesome/src/objects/tag.rs

```diff
@@ -34,8 +36,87 @@ impl Default for TagState {
 impl<'lua> Tag<'lua> {
     fn new(lua: &'lua Lua, args: Table) -> rlua::Result<Object<'lua>> {
         let class = class::class_setup(lua, "tag")?;
-        Ok(Tag::allocate(lua, class)?.handle_constructor_argument(args)?
-                                     .build())
+        Ok(object_setup(lua, Tag::allocate(lua, class)?)?
+                .handle_constructor_argument(args)?
+                .build())
+    }
+
+    pub fn get_clients(&self) -> rlua::Result<Table<'lua>> {
+        self.0.table()?.get("__clients")
+    }
+
+    pub fn set_clients(&mut self, clients: rlua::Table) -> rlua::Result<Value> {
+        // TODO: this is a really inefficient solution ( O(n*m) for the search )
+        //   Since it does not generally treat big arrays, this may be acceptable,
+        //   However a faster algorithm would not hurt if someone finds one
+
+        let prev_clients = self.get_clients()?;
+
+        // Indexes refering to prev_clients
+        let mut not_common_clients: Vec<u32> = (1..(prev_clients.len()? + 1) as u32)
+                                                    .collect();
+        let mut num_common_clients = 0;
+            
+        for client in clients.clone().sequence_values::<AnyUserData>() {
+            let mut idx = -1;
+            let client = Client::cast(client?.clone().into())?;
+            let self_ref = &*client.state()? as *const _;
+
+            for pair in prev_clients.clone().pairs::<Integer, AnyUserData>() {
+                let (i, c) = pair?;
+                if &*c.borrow::<ClientState>()? as *const _ == self_ref {
+                    idx = i;
+                    num_common_clients += 1;
+                    not_common_clients.remove((i - num_common_clients) as usize);
+                    break;
+                }
+            }
+            if idx != -1 {
+                // TODO: emit signal
+            }
+        }
+        for idx in not_common_clients.iter() {
+            // prev_clients.get::<Integer, ClientState>(idx)?
+            // TODO: emit signal and garbage if not referenced anymore
+        }
+        self.0.table()?.set("__clients", clients)?;
+        Ok(Value::Nil)
+    }
+
+    pub fn client_index(&self, client: &Client) -> rlua::Result<Integer> {
+        let clients = self.get_clients()?;
+        let self_ref = &*client.state()? as *const _;
+        for pair in clients.pairs::<Integer, AnyUserData<'lua>>() {
+            let (i, c) = pair?;
+            if &*c.borrow::<ClientState>()? as *const _ == self_ref {
```

Comment:
Can you not just cast the `client` to an `Object` to get to the `AnyUserData` to compare directly against the list of clients (and thus defeat the need for using this pointer comparison logic).

Reply:
But neither Object nor AnyUserData implements the Eq trait. One could change the object implementation, but it would come back to reference comparison, only inside the object. Unless you mean comparison by value, I can't see what you mean, sorry.